client/my-sites/stats/site.jsx

```diff
@@ -148,9 +149,15 @@ module.exports = React.createClass( {
 						queryDate={ queryDate }
 						period={ this.props.period }
 						chartTab={ this.state.chartTab } />
-					<DatePicker
-						period={ period }
-						date={ this.state.chartDate } />
+					<StatsPeriodNavigation
+						date={ this.props.date }
+						period={ this.props.period.period }
+						url={ `/stats/${ this.props.period.period }/${ site.slug }` }
```

Comment:
I'm not sure how I feel about this. It simplifies the props a lot, but I don't know exactly why but I feel that storing route params in the state is the right thing to do :). Maybe because they are ephemeral.

Thought, we already store `selectedSiteId` and `editedPostId`